source/val/BasicBlock.h

```diff
@@ -78,21 +78,26 @@ class BasicBlock {
   /// Returns true if the block is reachable in the CFG
   bool is_reachable() const { return reachable_; }
 
-  /// Returns the type of the BasicBlock
-  bool is_type(BlockType type) const { return type_.test(type); }
+  /// Returns true if BasicBlock is of the given type
+  bool is_type(BlockType type) const {
+    if (type == kBlockTypeUndefined) return type_.none();
+    return type_.test(type);
+  }
 
-  /// Sets the reacability of the basic block in the CFG
-  void set_reachability(bool reachability) { reachable_ = reachability; }
+  /// Sets the reachability of the basic block in the CFG
+  void set_reachable(bool reachability) { reachable_ = reachability; }
 
   /// Sets the type of the BasicBlock
-  void set_type(BlockType type) { type_.set(static_cast<int>(type)); }
+  void set_type(BlockType type) {
+    type_.set(type);
```

Comment:
This won't work if type is the Undefined type.  Either make that part of the contract (in the blurb) and perhaps assert; or make setting to Undefined clear all the bits.